ua-snap/serdp-fish-and-fire-webapp
# Move report to its own page and rework everything to use router with permalinks

Closes #59.
Closes #54 (incidentally!).

The primary goal of this PR is to add permalink support to allow users to share the URL to a specific AOI report directly. A few things changed along the way to make this possible, including:

- Before this PR, the AOI report "view" was essentially just a `<div v-if="...">` inside `index.vue`. This PR now makes the report view a proper Nuxt page and ties it into the routing system.
- Because the AOI report is now a proper page, your browser's back button works as you would expect. If you choose to use your brower's back button instead of the "Go Back" button on the AOI report page, the result is the same. It takes you back to the AOI search interface, which is now cleared of the previous search results (resolving #54).
- Some or all of the AOI polygon map jitters are gone because Nuxt is doing a fresh reload of the `Map.vue` component after the page transition instead of awkwardly resizing the map div on the same page as a new polygon was loading.
- To avoid horrendous permalinks like http://localhost:3000/report/Big%20Eldorado%20Creek-Goldstream%20Creek%20%28HUC12%3A%20190803090702%29 , I've modified the `csv2json.py` script to generate a 6-digit hash for each AOI name, and this is used as the unique identifier for permalinks. For example: http://localhost:3000/report/a49fa4. I considered using a numeric sequence instead (1, 2, 3...) but using a hash is a more failsafe solution that will not be impacted by subsequent CSV updates as long as the AOI names remain the same.

To test:

- Run through the new "Convert CSV files to JSON" README instructions in the `url_permalinks` branch to generate JSON files with 6-digit hash names and a new `map_hash.json` file that maps the 6-digit hashes to their human readable names. I've removed the superfluous `convert.sh` script and moved that stuff into `csv2json.py` itself because it was easier to generate the `hash_map.json` that way.
- Run the app, make sure you can select an AOI each of the three ways (dropdown menu, bulleted links of map click results, and polygon click), and make sure everything on the report page still works as expected.
